lms/services/grading_info.py

```diff
@@ -83,33 +73,31 @@ def get_students_for_grading(
 
         return students
 
-    def upsert_from_request(self, request):
+    def upsert(self, lti_user, lis_result_sourcedid, lis_outcome_service_url):
         """
         Update or create a record based on the LTI params found in the request.
 
         This function will do nothing if the correct parameters cannot be
-        found in the request.
-
-        :arg request: A pyramid request
+        found.
         """
-        try:
-            parsed_params = self._ParamsSchema(request).parse()
```

Comment:
This is IMO a very long winded way of validating two parameters.

It also doesn't do anything the the failed validation.